src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs

```diff
@@ -8019,10 +8019,10 @@ unsafe public static void Main()
             CreateStandardCompilation(text, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
                 // (7,23): error CS0213: You cannot use the fixed statement to take the address of an already fixed expression
                 //       fixed (int *j = &i) { }  // CS0213
-                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "&i"),
-                // (14,26): error CS0213: You cannot use the fixed statement to take the address of an already fixed expression
+                Diagnostic(ErrorCode.ERR_FixedNotNeeded, "&i").WithLocation(7, 23),
+                // (14,26): error CS9365: The given expression cannot be used in a fixed statement
```

Comment:
This error message seems worse (less specific) than the previous one. Why did it change?

Reply:
The old message was actually incorrect. "already fixed" would apply if `&` was used.  Here we do not use `&` . It might actually work if there is an extension method of right shape., but most likely it is just a typo.

I do not think the new message is worse than the old one. 

---
In reply to: [167710934](https://github.com/dotnet/roslyn/pull/24469#discussion_r167710934) [](ancestors = 167710934)